src/azure-cli-core/azure/cli/core/profiles/_shared.py

```diff
@@ -5,7 +5,57 @@
 
 # TODO Move this to a package shared by CLI and SDK
 
+from importlib import import_module
+from enum import Enum
+
+
+class APIVersionException(Exception):
+    # API Version is not available for API profile or resource type.
+    pass
+
+
+class ResourceType(Enum):  # pylint: disable=too-few-public-methods
+    def __init__(self, type_name, sdk_module, client_name):
+        self.type_name = type_name
+        self.sdk_module = sdk_module
+        self.client_name = client_name
+
+    MGMT_STORAGE_STORAGE_ACCOUNTS = ('Microsoft.Storage/storageAccounts',
+                                     'azure.mgmt.storage',
+                                     'StorageManagementClient')
+
+
 AZURE_API_PROFILES = {
-    '2016-01-01-example': {},
-    '2015-06-15-example': {}
+    'latest': {
+        ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS: '2016-12-01'
+    },
+    '2016-example': {
+        ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS: '2016-12-01'
+    },
+    '2015-example': {
+        ResourceType.MGMT_STORAGE_STORAGE_ACCOUNTS: '2015-06-15'
+    }
```

Comment:
> So all of the profiles are statically hard-coded in the CLI?

It would go in a shared package that's used by both the CLI and the Python SDK.

Yes it's decoupled. Everything in the `shared.py` file will eventually be moved out to a shared package to be used by both the CLI and SDK.